app/pods/publications/publication/documents/controller.js

```diff
@@ -57,4 +83,131 @@ export default class PublicationDocumentsController extends Controller {
+    const savePromises = this.newPieces.map(async(piece) => {
+      try {
+        this.savePiece.perform(piece);
+      } catch (error) {
+        this.deletePiece.perform(piece);
+        throw error;
+      }
+    });
+    yield all(savePromises);
+    this.isOpenPieceUploadModal = false;
+    this.newPieces = A();
+  }
+
+  /**
+   * Save a new document container and the piece it wraps
+   */
+  @task
+  *savePiece(piece) {
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.save();
+    yield piece.save();
+    const pieces = yield this.model.case.hasMany('pieces').reload();
+    pieces.pushObject(piece);
+    yield this.model.case.save();
+  }
+
+  @task
+  *cancelUploadPieces() {
+    const deletePromises = this.newPieces.map((piece) => this.deletePiece.perform(piece));
+    yield all(deletePromises);
+    this.newPieces = A();
+    this.isOpenPieceUploadModal = false;
+  }
+
+  @task
+  *deletePiece(piece) {
+    const file = yield piece.file;
+    yield file.destroyRecord();
+    this.newPieces.removeObject(piece);
+    const documentContainer = yield piece.documentContainer;
+    yield documentContainer.destroyRecord();
+    yield piece.destroyRecord();
+  }
+
+
+  /** TRANSLATION ACTIVITIES **/
+
+  @action
+  openTranslationRequestModal() {
+    this.translateActivity.finalTranslationDate = this.model.publicationFlow.translateBefore;
+    this.translateActivity.pieces = this.selectedPieces;
+    this.showTranslationModal = true;
+  }
+
+  get getTranslateActivityBeforeDate() {
+    if (this.model.publicationFlow.translateBefore) {
+      return this.model.publicationFlow.translateBefore;
+    }
+    return new Date();
+  }
+
+  @action
+  async saveTranslationActivity() {
+    this.showLoader = true;
+    this.showTranslationModal = false;
+    const translateSubCaseType = EmberObject.create({
+      id: CONFIG.SUBCASE_TYPES.vertalen.id,
+      uri: CONFIG.SUBCASE_TYPES.vertalen.url,
+    });
+
+    // TODO take from other subcase maybe?
+    const shortTitle = await this.model.case.shortTitle;
+    const title = await this.model.case.title;
+
+    // Create subase.
+    const subcase = await this.subcasesService.createSubcaseForPublicationFlow(this.model.case, this.model.publicationFlow, translateSubCaseType, shortTitle, title);
+
+    // Create activity in subcase.
+    await this.activityService.createNewTranslationActivity(this.translateActivity.finalTranslationDate, this.translateActivity.mailContent, this.translateActivity.pieces, subcase);
+
+    // Visual stuff.
+    this.showLoader = false;
+    this.currentPieces.forEach((piece) => {
+      piece.selected = false;
+    });
+    this.currentPieces = [...this.currentPieces];
+
+    // Reset local activity to empty state.
+    this.translateActivity = {
+      mailContent: '',
+      finalTranslationDate: '',
+      pieces: [],
+    };
+  }
+
+  @action
+  cancelTranslationModal() {
+    set(this.translateActivity, 'mailContent', '');
```

Comment:
Kan dit niet via `this.translateActivity.mailContent = ''`' als je de translate activity hebt en mailcontent is een property dat rechtstreeks op het model zit, denk ik dat je het gewoon in een normale object syntax kan schrijven.. daarna gewoon niet vergeten te saven.

Wordt het model hier automatisch gesaved en gerefreshed door de `set` functie te gebruiken?

Reply:
nope. Ember gooit dan fouten. Moet een set zijn.
Is niet echt een model, is een tracked array.